martinLi/assignments/treesBFS/lc637/lc637.js

```diff
@@ -0,0 +1,42 @@
+// Problem Statement #
+
+// Given a binary tree, populate an array to represent the averages of all of its levels.
+
+class TreeNode {
+  constructor(value) {
+    this.value = value;
+    this.left = null;
+    this.right = null;
+  }
+}
+
+const find_level_averages = function (root) {
+  //leetcodesolnbfs
+  //get back to this
+  let result = [],
+    q = [root]; // using a queue
+  while (q.length) {
+    // while there is stuff in the queue
+    let qlen = q.length,
+      row = 0;
+    for (let i = 0; i < qlen; i++) {
+      let curr = q.shift(); //removes element
+      row += curr.value;
```

Comment:
comments as you go along should not take the place of a great high level overview section. If you need inline comments while you write your program please make sure to remove them before submitting